File: src/main/java/io/github/samarium150/minecraft/mod/structures_compass/init/CommandRegistry.java
package io.github.samarium150.minecraft.mod.structures_compass.init;

import io.github.samarium150.minecraft.mod.structures_compass.command.*;
import net.minecraftforge.event.RegisterCommandsEvent;
import net.minecraftforge.eventbus.api.SubscribeEvent;
import net.minecraftforge.fml.common.Mod;

import javax.annotation.Nonnull;

/**
 * The registry of commands in the mod
 */
@Mod.EventBusSubscriber
public final class CommandRegistry {
    
    private CommandRegistry() { }
    
    /**
     * Register commands
     * @param event RegisterCommandsEvent
     * @see RegisterCommandsEvent
     */
    @SuppressWarnings("deprecation")
    @SubscribeEvent
    public static void register(@Nonnull final RegisterCommandsEvent event) {
        GetCompass.register(event.getDispatcher());
    }
}

File: src/main/kotlin/io/github/samarium150/minecraft/mod/structures_compass/init/CommandRegistry.kt
/*
 * Copyright (c) 2022 Samarium
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/gpl-3.0.html>.
 */
package io.github.samarium150.minecraft.mod.structures_compass.init

import io.github.samarium150.minecraft.mod.structures_compass.server.command.GetCompass
import net.fabricmc.fabric.api.command.v1.CommandRegistrationCallback

object CommandRegistry {
    fun init() {
        CommandRegistrationCallback.EVENT.register { dispatcher, _ ->
            run {
                GetCompass.register(dispatcher)
            }
        }
    }
}
